src/Admin/AbstractAdmin.php

```diff
@@ -3262,24 +3266,43 @@ public function canAccessObject($action, $object)
     }
 
     /**
-     * @return string[]
+     * @return MutableTemplateRegistryInterface
      */
-    protected function configureExportFields(): array
+    final public function getTemplateRegistry()
     {
-        return $this->getModelManager()->getExportFields($this->getClass());
+        // NEXT_MAJOR: Remove the deprecation and uncomment the exception.
+        if (!$this->hasTemplateRegistry()) {
+            @trigger_error(sprintf(
+                'Calling %s() when there is no template registry is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw an exception in 4.0.'
+                .' Use %s::hasTemplateRegistry() to know if the template registry is set.',
+                __METHOD__,
+                __CLASS__
+            ), E_USER_DEPRECATED);
+        }
+        //if (false === $this->hasTemplateRegistry()) {
+        //    throw new \LogicException(sprintf('Unable to find the template registry for admin `%s`.', static::class));
+        //}
+
+        return $this->templateRegistry;
     }
 
-    protected function configureQuery(ProxyQueryInterface $query): ProxyQueryInterface
+    final public function hasTemplateRegistry(): bool
```

Comment:
Is this method used outside the `AbstractAdmin`? If so, should be added to the interface?

Reply:
It is new method, so it is not used for now. IMO all `setTemplateRegistry`, `getTemplateRegistry` and `hasTemplateRegistry` should be add to seperate `TemplateRegistryOwnerInterface`. `AdminInterface` should inplemented it.